crates/chain/src/chain_oracle.rs

```diff
@@ -17,7 +17,7 @@ pub trait ChainOracle {
     fn is_block_in_chain(
         &self,
         block: BlockId,
-        chain_tip: BlockId,
+        chain_tip: Option<&BlockId>,
```

Comment:
Mh, wouldn't it be better if we just had `BlockId` here? I can't really think of any case where someone would call `is_block_in_chain(block, None)`, giving that it would always return `None`